### PR TITLE
fix: Fix STFC email search

### PR DESCRIPTION
**Pull request by simonfernandes**
## Description
(View without whitespace changes)

This is a hotfix that fixes an error when searching for a user to add as PI or Co-I, when the UOWS returns no results.

Refs UserOfficeProject/user-office-project-issue-tracker#632

## How Has This Been Tested

- Tested that a frontend error displays rather than the backend throwing an error
- Fix aligns with the backend error thrown

### CODE REVIEW

**Review by TickleThePanda — CHANGES_REQUESTED** [review]
Any chance of another unit test here for the expected behaviour from the UOWS?

File: src/datasources/stfc/StfcUserDataSource.ts [hunk 1]
```diff
@@ -184,7 +184,7 @@ export class StfcUserDataSource implements UserDataSource {
     const uowsRequest = searchableOnly
       ? client.getSearchableBasicPersonDetailsFromEmail(token, email)
       : client.getBasicPersonDetailsFromEmail;
```
Comment (TickleThePanda) [inline src/datasources/stfc/StfcUserDataSource.ts:186]: I'm a bit confused about how the `client.getBasicPersonDetailsFromEmail` would work, too, as it doesn't look like it's being called to me?

**Comment by simonfernandes** [comment]
The tests need reviewing with whitespace hidden

**Review by mr-field — COMMENTED** [review]
The tests look good, I just have a minor improvement to suggest. Could you also add a test for when `getBasicUserDetailsByEmail` is called with an invalid token, for completeness?

**Comment by simonfernandes** [comment]
I've added a test for invalid token... I think. I wasn't quite sure how to do it which is why I left it out. I think this covers it. The soap client method catches the exception and returns the result, which would be an object with null return value.

File: src/datasources/stfc/StfcUserDataSource.spec.ts [hunk 2]
```diff
@@ -1,61 +1,123 @@
...
+    mockEnsureDummyUserExists.mockClear();
+  });
 
-  const roles = await userdataSource.getUserRoles(dummyUserNumber);
+  test('When getting basic user details, the basic user is created and returned', async () => {
+    const result = await userdataSource.getBasicUserDetailsByEmail('valid');
 
-  return expect(roles).toHaveLength(0);
+    expect(mockGetBasicPersonDetailsFromEmail).toBeCalledTimes(0);
+    expect(mockGetSearchableBasicPersonDetailsFromEmail).toBeCalled();
```
[the remark this answers is not in the document]
Comment (simonfernandes) [reply src/datasources/stfc/StfcUserDataSource.spec.ts]: Thanks, I couldn't find the answer to that easily either, and it probably reads better too.

**Review by mr-field — APPROVED** [review]

**Review by vyshnavi1605 — APPROVED** [review]

**Pull request merged**